packages/core/src/components/way-autosuggest/way-autosuggest.tsx

```diff
@@ -0,0 +1,265 @@
+    if (!this.localSelected) return;
+
+    this.filteredOptions = this.filteredOptions.map(fOption => ({
+      ...fOption,
+      selected: !!this.localSelected.find(selected => selected.value === fOption.value),
+    }));
+  }
+
+  /**
+   * Listener when user adds an option.
+   * @param {ISelectOption} option
+   */
+  private optionSelectedListener(option: ISelectOption) {
+    if (!this.localConfig.multi) {
+      this.localSelected = [option];
+      this.unFocus();
+    } else if (!option.selected) {
+      this.localSelected.push(option);
+    } else {
+      this.localSelected = this.localSelected.filter(selected => selected.value !== option.value);
+    }
+    this.updateOptions();
+  }
+
+  /**
+   * Listener when user removes a tag.
+   * @param {ISelectOption} option
+   */
+  private removeTagListener(option) {
+    if (!this.localSelected) return;
+
+    this.localSelected = this.localSelected.filter(selectedOption => selectedOption.value != option.value);
+    this.updateOptions();
+  }
+
+  private handleOnInput = (evt: InputEvent) => {
+    const input = evt.target as HTMLInputElement;
+
+    this.hasFocus = true;
+    this.inputValue = input.value;
+    this.updateOptions();
+  };
+
+  private validationErrors(): string[] {
+    return this.validation ? this.validation(this.localSelected) : [];
+  }
+
+  private calculateMinimumInputWidth(): string {
+    if (!this.inputEl) return '7em';
+
+    const calculation = this.inputEl.value.length * 0.7;
+    if (calculation < 7) {
+      return '7em';
+    } else if (calculation < 15) {
+      return calculation + 'em';
+    } else {
+      return '99%';
+    }
+  }
+
+  private renderTags() {
+    if (!this.localSelected) {
+      return null;
+    } else if (this.localSelected.length < this.localConfig.maxTags + 1) {
+      return (
+        <this.Fragment>
+          {this.localSelected.length > 0 &&
+            this.localSelected.map(tag => (
+              <div title={tag.label} onClick={() => this.removeTagListener(tag)} class="tag" style={{ backgroundColor: this.config.tagColor }}>
+                <span>{tag.label}</span>
+              </div>
+            ))}
+        </this.Fragment>
+      );
+    } else {
+      const text = `${this.localSelected.length} ${this.localConfig.selectedText}`;
+      return (
+        <div class="tag counter" title={text} style={{ backgroundColor: this.config.tagColor }}>
+          <span>{text}</span>
+        </div>
+      );
+    }
+  }
+
+  /**
+   * Used to render multiple HTMLElements without using a wrapper element.
+   * In React JSX we can use '<> <div/><div/><div/> </>' but is not allowed in TS for StencilJS.
+   * 
+   * @param { any } _
+   * @param { HTMLElement[] } children 
+   * 
+   * @returns HTMLElement[]
+   */
+  private Fragment = (_: any, children: HTMLElement[]): HTMLElement[] => [ ...children ];
+
+  private renderInputOutsideShadowRoot(container: HTMLElement, name: string, value: string | null) {
```

Comment:
I would like this function in a util class (like Ionic does), because we will need this function in all form components.